src/Show.jl

```diff
@@ -326,21 +326,29 @@ q - 1
 
 ```
 """
-function nrchars(t::CharTable, char::Int64)
-	if char > irrchartypes(t)
-		throw(DomainError(char, "Cannot calculate number of characters in reducible types."))
-	else
-		o=t.ring(1)
-		result=t[char].sum(o//o)
-		return shrink(result)
-	end
-end
-function nrchars(t::SimpleCharTable, char::Int64)
-	if char > chartypes(t)
-		throw(DomainError(char, "Character type is out of range."))
-	end
-	1
+nrchars(t::Table, char::Int64) = nrchars(t[char])
+
+@doc raw"""
+    nrchars(chi::GenericCharacter)
+
+Return the number of characters in the generic character `chi`.
+
+# Examples
+```jldoctest
+julia> g=genchartab("GL2");
+
+julia> nrchars(g[1])
+q - 1
+
+```
+"""
+function nrchars(chi::GenericCharacter)
+    t = parent(chi)
+    o=CycloSum(t.modulusring(1), t.argumentring(0))
```

Comment:
I think something went wrong here. We don't have `CycloSum`, `t.modulusring` and `t.argumentring` anymore.

Reply:
This is just a rebase of an old commit, I didn't test it afterwards. Anyway, your new PR #157 supersedes this :-)